app/models/financialDetails/responses/ChargesResponse.scala

```diff
@@ -46,6 +46,10 @@ case class ChargesResponse(balanceDetails: BalanceDetails,
         _.mainType
       )
 
+      val mainTransaction = financialDetails.find(_.transactionId.equals(document.transactionId)).flatMap(
```

Comment:
Please add type definition for mainTransaction val, this easy up reading this code